### DISCUSSION TITLE
How to bind donburi entity data to the furex components

### DISCUSSION THREAD

**Discussion by phalasz** [yohamta/furex#25] [Q&A]
Hello,

I'm working on a roguelike in go using your excellent donburi ecs and would like to use furex to render the UI for my game.

I was wondering wether you might have any pointers on how I could for example bind my player's stats to a UI component.

As a small example on what I would like to achieve:
I have a player entity with a stats component that has the player's current and max health.
I would like to create a progress bar/stats bar UI component to display this information.
I can draw the component with static data, but I am not sure how I could reference the player's stats component without having 
a donburi query attached to the UI component it self and use that to grab the data in the update handler.

Is there a preferred way to achieve this or shall I use donburi queries which would work in case of the player as there is alway one, but for enemies this might not work.

Any pointers would be much appreciated.

**Comment by phalasz** [comment]
Actually, I think you already made this simpler in 1.3 of donburi with the event bus. I have missed that update it seem.

**Comment by yohamta0** [comment]
Hello,
Thank you for using donburi and furex!

I also use them in my own games.
For binding values, I use normal functions to update the view's values. Using the event bus feature for binding values sounds very nice to me. Maybe we could make a small example with that.

Though I am still wondering how we can implement UI  on top of ECS architecture. The UI world and ECS world in my game are separated for now, so a bit cumbersome to bind values between those.

**Reply by phalasz** [reply]
I'll see if I can post something when I have things working as I want them.
